ui/admin/tests/acceptance/targets/injected-application-credential-sources-test.js

```diff
@@ -0,0 +1,437 @@
+import { visit, find, findAll, click, currentURL } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';
+import { Response } from 'miragejs';
+import a11yAudit from 'ember-a11y-testing/test-support/audit';
+import { authenticateSession } from 'ember-simple-auth/test-support';
+
+module(
+  'Acceptance | targets | injected application credential sources',
+  function (hooks) {
+    setupApplicationTest(hooks);
+    setupMirage(hooks);
+
+    let getCredentialLibraryCount;
+    let getCredentialCount;
+    let credentialSourceCount;
+    let randomlySelectedCredentialLibraries;
+    let randomlySelectedCredentials;
+
+    const instances = {
+      scopes: {
+        global: null,
+        org: null,
+        project: null,
+      },
+      vaultCredentialStore: null,
+      staticCredentialStore: null,
+      credentialLibraries: null,
+      credentialLibrary: null,
+      credentials: null,
+      credential: null,
+    };
+    const urls = {
+      globalScope: null,
+      orgScope: null,
+      projectScope: null,
+      targets: null,
+      target: null,
+      credentialLibraries: null,
+      credentialLibrary: null,
+      addInjectedApplicationCredentialSources: null,
+      injectedApplicationCredentialSources: null,
+    };
+
+    hooks.beforeEach(function () {
+      // Generate resources
+      instances.scopes.global = this.server.create('scope', { id: 'global' });
+      instances.scopes.org = this.server.create('scope', {
+        type: 'org',
+        scope: { id: 'global', type: 'global' },
+      });
+      instances.scopes.project = this.server.create('scope', {
+        type: 'project',
+        scope: { id: instances.scopes.org.id, type: 'org' },
+      });
+      instances.vaultCredentialStore = this.server.create('credential-store', {
+        type: 'vault',
+        scope: instances.scopes.project,
+      });
+      instances.staticCredentialStore = this.server.create('credential-store', {
+        type: 'static',
+        scope: instances.scopes.project,
+      });
+      instances.credentials = this.server.createList('credential', 2, {
+        scope: instances.scopes.project,
+        credentialStore: instances.staticCredentialStore,
+      });
+      instances.credentialLibraries = this.server.createList(
+        'credential-library',
+        2,
+        {
+          scope: instances.scopes.project,
+          credentialStore: instances.vaultCredentialStore,
+        }
+      );
+      instances.credentialLibrary = instances.credentialLibraries[0];
+      instances.credential = instances.credentials[0];
+      instances.target = this.server.create('target', {
+        scope: instances.scopes.project,
+        type: 'ssh',
+      });
+
+      randomlySelectedCredentials = this.server.schema.credentials
+        .all()
+        .models.filter(Boolean)
+        .map((cred) => cred.id);
```

Comment:
What does the `.models` do? Doesn't `.all()` return models?

Would this work? 
```suggestion
      randomlySelectedCredentials = this.server.schema.credentials
        .where(Boolean)
        .map((cred) => cred.id);
```

Reply:
ah, I believe this is a copy/paste error. let me clean this up

Reply:
I removed the filter part as that is not needed